pkg/chaosctl/cmd/debug.go

```diff
@@ -68,104 +57,61 @@ Examples:
-		},
-		SilenceErrors: true,
-		SilenceUsage:  true,
-		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
-			clientset, err := common.InitClientSet()
-			if err != nil {
-				return nil, cobra.ShellCompDirectiveDefault
-			}
-			if len(args) != 0 {
-				return nil, cobra.ShellCompDirectiveNoFileComp
-			}
-			return listChaos(networkChaos, o.namespace, toComplete, clientset.CtrlCli)
-		},
+	for chaosType, debug := range debugs {
+		debugCmd.AddCommand(debugResourceCommand(o, chaosType, debug))
 	}
 
-	stressCmd := &cobra.Command{
-		Use:   `stresschaos (CHAOSNAME) [-n NAMESPACE]`,
-		Short: `Print the debug information for certain stress chaos`,
-		Long:  `Print the debug information for certain stress chaos`,
-		RunE: func(cmd *cobra.Command, args []string) error {
-			clientset, err := common.InitClientSet()
-			if err != nil {
-				return err
-			}
-			return o.Run(stressChaos, args, clientset)
-		},
-		SilenceErrors: true,
-		SilenceUsage:  true,
-		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
-			clientset, err := common.InitClientSet()
-			if err != nil {
-				return nil, cobra.ShellCompDirectiveDefault
-			}
-			if len(args) != 0 {
-				return nil, cobra.ShellCompDirectiveNoFileComp
-			}
-			return listChaos(stressChaos, o.namespace, toComplete, clientset.CtrlCli)
-		},
-	}
+	debugCmd.PersistentFlags().StringVarP(&o.namespace, "namespace", "n", "default", "namespace to find chaos")
+	err := debugCmd.RegisterFlagCompletionFunc("namespace", func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		client, cancel, err := common.CreateClient(context.TODO(), managerNamespace, managerSvc)
+		if err != nil {
+			logger.Error(err, "fail to create client")
+			return nil, cobra.ShellCompDirectiveNoFileComp
+		}
+		defer cancel()
+
+		completion, err := client.ListNamespace(context.TODO())
+		if err != nil {
+			logger.Error(err, "fail to complete resource")
+			return nil, cobra.ShellCompDirectiveNoFileComp
+		}
 
-	ioCmd := &cobra.Command{
-		Use:   `iochaos (CHAOSNAME) [-n NAMESPACE]`,
-		Short: `Print the debug information for certain io chaos`,
-		Long:  `Print the debug information for certain io chaos`,
+		return completion, cobra.ShellCompDirectiveNoFileComp | cobra.ShellCompDirectiveNoSpace
+	})
+	return debugCmd, err
+}
+
+func debugResourceCommand(option *DebugOptions, chaosType string, debug debug.Debug) *cobra.Command {
+	return &cobra.Command{
+		Use:   fmt.Sprintf(`%s (CHAOSNAME) [-n NAMESPACE]`, chaosType),
+		Short: fmt.Sprintf(`Print the debug information for certain %s`, chaosType),
+		Long:  fmt.Sprintf(`Print the debug information for certain %s`, chaosType),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			clientset, err := common.InitClientSet()
+			ctrlclient.DisableRuntimeErrorHandler()
```

Comment:
Now, we only call `ctrlclient.DisableRuntimeErrorHandler()` in the port forwarder as this function is used to prevent complaining from port forwarder.